src/js/express.js

```diff
@@ -0,0 +1,179 @@
+    if (!that.options.config || !that.options.config.express) {
+        console.error("Cannot initialize express because you have not supplied a 'config' option.");
+        return;
+    }
+
+    var express  = require("express");
+    that.express = express();
+
+    // Wire together all routers and components, beginning with ourselves
+    for (var a = 0; a < that.options.members.directChildrenOfInterest.length; a++) {
+        var directChildNickname = that.options.members.directChildrenOfInterest[a];
+        var childComponent      = that[directChildNickname];
+        if (fluid.hasGrade(childComponent.options, "gpii.express.router")) {
+            // The router has to wire its own paths to preserve methods and path variables.
+            // We just "use" it at the root level, and let it do the rest.
+            that.express.use("/", childComponent.options.router);
+        }
+        else if (fluid.hasGrade(childComponent.options, "gpii.express.middleware")) {
+            that.express.use(childComponent.getMiddlewareFunction());
+        }
+
+        // Recurse from here on down.
+        that.connectDirectDescendants(childComponent, directChildNickname);
+    }
+
+    that.express.set("port", that.options.config.express.port);
+    that.server = that.express.listen(that.options.config.express.port, function(){
+        console.log("Express server listening on port " + that.express.get("port"));
+
+        console.log("Express started...");
+        that.events.started.fire(that.express);
+    });
+};
+
+gpii.express.stopServer = function(that) {
+    that.server.close(function(){
+        console.log("Express stopped...");
+        that.events.stopped.fire();
+    });
+};
+
+fluid.defaults("gpii.express", {
+    gradeNames: ["fluid.standardRelayComponent", "autoInit"],
+    members: {
+        directChildrenOfInterest: [],
+        childrenByParent:         {}
+    },
+    path: "/",
+    express: null,
+    distributeOptions:[
+        {
+            record: {
+                "funcName": "gpii.express.registerComponentLineage",
+                "args": ["{arguments}.0", "{gpii.express}"]
+            },
+            target: "{that gpii.express.router}.options.listeners.routerLoaded"
+        },
+        {
+            record: {
+                "funcName": "gpii.express.registerComponentLineage",
+                "args": ["{arguments}.0", "{gpii.express}"]
+            },
+            target: "{that gpii.express.middleware}.options.listeners.onCreate"
+        }
+    ],
+    invokers: {
+        "connectDirectDescendants" : {
+            "funcName": "gpii.express.connectDirectDescendants",
+            "args":     ["{that}", "{arguments}.0", "{arguments}.1"]
+        }
+    },
+    events: {
+        started:  null,
+        stopped:  null
+    },
+    listeners: {
+        onCreate: [
+            {
+                "funcName": "gpii.express.init",
+                "args": "{that}"
+            }
+        ],
+        onDestroy: [
```

Comment:
Namespace these for their function, so that an integrator could override the individual handlers - 
e.g. "onDestroy.stopServer": {
   funcName: ......
etc
You can also leave out the array containment here since there is just a single listener